src/main/java/core/basesyntax/dao/DaoManufacturerImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class DaoManufacturerImpl implements DaoManufacturer {
+    @Override
+    public Manufacturer create(Manufacturer item) {
+        Storage.manufacturers.put(Storage.generateId(), item);
```

Comment:
let's create method save() in Storage class and use it here for generating and saving new data. example:
```
Storage.saveManufacturer(manufacturer); // this method will create id and call put() method from map
```